### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by jgrund**
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2112)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  push:
+    branches: [master, b6.1]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 12 * * 4'
+
+jobs:
+  analyse:
```
Comment (johnsonw) [inline .github/workflows/codeql-analysis.yml:13]: Should this be `analyze`?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      with:
+        languages: javascript, python
```
Comment (johnsonw) [inline .github/workflows/codeql-analysis.yml:35]: I'm guessing rust isn't an option?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (jgrund) [reply .github/workflows/codeql-analysis.yml:13]: No

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (jgrund) [reply .github/workflows/codeql-analysis.yml:35]: Correct

**Pull request merged**